docs/dice.md

```diff
@@ -0,0 +1,170 @@
+
+While the core concepts are unchanged, there are a few notable differences from TCG DICE, mainly around the encoding of payloads.  See the [Open DICE Formats](#open-dice-formats) section for the details.
+
+## DICE Concepts
+
+### Glossary
+
+* __DICE Hardware RoT__ - The HW implementation of the DICE processing engine, acting as the terminal trust anchor in a DICE device.
+
+* __Unique Device Secret (UDS)__ - A permanent, (statistically) unique, secret value.  Every instance of a DICE device has its own UDS stashed away in a location only accessible to the DICE engine.  The UDS can be injected at manufacturing time, or generated by the device itself using a suitable PRNG.
+
+* __Trusted Computing Base (TCB)__ : All the security relevant HW and SW in a DICE device.  DICE organizes the TCB in multiple "components", which are grouped in "layers" stacked one on top of the other, starting with the DICE engine.  A layer boundary represents an explicit handover point from one component to one or more components in the "next" stage of the boot chain.
+
+* __TCB Component Identifier (TCI)__ - A (statistically) unique identifier for a TCB component.  This value is computed by digesting the component's measurement along with other component-specific attributes, e.g., vendor name, version, etc.
+
+* __Compound Device Identifier (CDI)__ - This value represents the (statistically) unique identity of each layer in the bootstrap sequence.  The CDI for a given layer is obtained by hashing together the predecessor CDI and the TCI for the current layer.  This recursive construction is such that the resulting CDIs are inextricably chained: when one CDI changes (e.g., because the measured software component is updated, and the TCI mutates accordingly), all its successor CDIs change as well.  Crucially, the CDI is directly used as a seed for deriving cryptographic credentials (e.g., an ECDSA key pair) for the corresponding TCB component.
+
+> NOTE: At first it might seem like the scalability properties of DICE verification vis-a-vis software & configuration updates of TCB components are terrible.  The verifier would have to be re-provisioned with all the updated identities of all the devices of the same "class" if a bit was flipped in their TCB.  Luckily, the situation is not so gloomy: as long as the lowest layer in the DICE chain is stable (as in "never updated") and the DICE compound identity is provided as a certification chain rooted either in the lowest layer or in an external CA that issues the cert for the lowest layer, an identity update of the i-th layer can be re-computed dynamically by a verifier without need to further interact with the supply chain.  The key requirement is the stability of the so-called "Layer 0" / "First Mutable Code" (FMC), and therefore of its cryptographic identity.  The latter is a pretty important bit in DICE and it's given the proper name of "Device Id".  The Device Id key pair is associated with an IEEE 802.1AR IDevID cert.
```

Comment:
The verification process is more practical (accurate?) if the certs contain the Digest/Hash info for the current layer. These are currently marked as optional in both specs. Can we add some text that makes a recommendation for implementations to include these entries?

Reply:
```suggestion
> NOTE: At first it might seem like the scalability properties of DICE verification vis-a-vis software & configuration updates of TCB components are terrible.  The verifier would have to be re-provisioned with all the updated identities of all the devices of the same "class" if a bit was flipped in their TCB.  Luckily, the situation is not so gloomy: as long as the lowest layer in the DICE chain is stable (as in "never updated") and the DICE compound identity is provided as a certification chain rooted either in the lowest layer or in an external CA that issues the cert for the lowest layer, an identity update of the i-th layer can be re-computed dynamically by a verifier without need to further interact with the supply chain -- except for re-provisioning the verifier with the updated measurement.  The key requirement is the stability of the so-called "Layer 0" / "First Mutable Code" (FMC), and therefore of its cryptographic identity.  The latter is a pretty important bit in DICE and it's given the proper name of "Device Id".  The Device Id key pair is associated with an IEEE 802.1AR IDevID cert.
```

Reply:
> The verification process is more practical (accurate?) if the certs contain the Digest/Hash info for the current layer. These are currently marked as optional in both specs. Can we add some text that makes a recommendation for implementations to include these entries?

I don't think we can make any recommendation from here.  We are just sort of passively documenting what happens.  What the text was missing was acknowledging the fact that the measurement (e.g., hash, version, etc.) is an integral part of the verification, not just the chaining of the certs.

Reply:
OK, good point, just my wishful thinking!